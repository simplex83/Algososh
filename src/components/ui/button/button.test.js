import { Button } from "./button";
import renderer from 'react-test-renderer';
import { render, screen, fireEvent } from '@testing-library/react';

describe("Button", () => {
    it("button is empty", () => {
        const button = renderer.create(<Button />).toJSON();
        expect(button).toMatchSnapshot();
    });
    it("button with text", () => {
        const button = renderer.create(<Button text='text' />).toJSON();
        expect(button).toMatchSnapshot();
    });
    it("button is disabled", () => {
        const button = renderer.create(< Button disabled />).toJSON();
        expect(button).toMatchSnapshot();
    });
    it("button is loader", () => {
        const button = renderer.create(< Button isLoader />).toJSON();
        expect(button).toMatchSnapshot();
    });
    it("button click", () => {
        window.alert = jest.fn();
        render(<Button text="text" onClick={() => alert("callback")} />);
        const button = screen.getByText("text");
        fireEvent.click(button);
        expect(window.alert).toHaveBeenCalledWith("callback");
    });
})