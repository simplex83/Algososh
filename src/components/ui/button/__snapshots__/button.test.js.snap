// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button button is disabled 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Button button is empty 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Button button is loader 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`Button button with text 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    text
  </p>
</button>
`;
